src/cairoWriter/writers/cairoAssertWriter.ts

```diff
@@ -1,20 +1,15 @@
 import { ASTWriter, SrcDesc } from 'solc-typed-ast';
 import { CairoAssert } from '../../ast/cairoNodes';
 import { CairoASTNodeWriter } from '../base';
-import { INDENT } from '../utils';
 
 export class CairoAssertWriter extends CairoASTNodeWriter {
   writeInner(node: CairoAssert, writer: ASTWriter): SrcDesc {
-    const assertExpr = `assert ${writer.write(node.vExpression)} = 1;`;
+    const expression = writer.write(node.vExpression);
 
     if (node.assertMessage === null) {
-      return [assertExpr];
+      return [`assert( ${expression} = 1 );`];
     } else {
-      return [
-        [`with_attr error_message("${node.assertMessage}"){`, `${INDENT}${assertExpr}`, `}`].join(
-          '\n',
-        ),
-      ];
+      return [`assert( ${expression} = 1, "${node.assertMessage}" );`];
```

Comment:
Booleans are available in cairo 1

Reply:
Similarly to https://github.com/NethermindEth/warp/pull/924/files#r1120567164 `= 1` should be removed and encoded on node level

Reply:
Regarding booleans, it might be necessary to update generated/warplib functions though. We'll have to think of that. Maybe it's better to stick with the old integer based comparisons and postpone using booleans.

Reply:
Ye, its a good idea to put it in other pr after this one is merged